### PR TITLE
Refactor some conditional expressions

### PR DESCRIPTION
**Pull request by YotaroSeki**
Hi!
I've refactored some conditional expressions as below.

- ```and``` -> ```&&```
- ```or``` -> ```||```
- ```not``` -> ```!```
- ```hoge == ""``` -> ```hoge.blank?```
- ```hoge == 0``` -> ```hoge.zero?```

Please review this PR.
Thank you.

### CODE REVIEW

**Comment by YotaroSeki** [comment]
@mshibuya 
Please review this PR.

File: lib/carrierwave/mounter.rb [hunk 1]
```diff
@@ -38,7 +38,7 @@ def uploaders
     end
 
     def cache(new_files)
-      return if not new_files or new_files == ""
+      return if !new_files || new_files.blank?
```
Comment (mshibuya) [inline lib/carrierwave/mounter.rb]: `!new_files` is unnecessary, because `nil.blank?` is true.

File: lib/carrierwave/uploader/url.rb [hunk 2]
```diff
@@ -15,8 +15,8 @@ module Url
       # [String] the location where this file is accessible via a url
       #
       def url(options = {})
-        if file.respond_to?(:url) and not (tmp_url = file.url).blank?
-          file.method(:url).arity == 0 ? tmp_url : file.url(options)
+        if file.respond_to?(:url) && !file.url.blank?
+          file.method(:url).arity.zero? ? file.url : file.url(options)
```
Comment (mshibuya) [inline lib/carrierwave/uploader/url.rb]: `tmp_url` should not be removed, because `#url` might be costly operation when using fog.

Comment (YotaroSeki) [reply lib/carrierwave/uploader/url.rb]: I didn't know ```#url``` is costly when using fog.
I added ```tmp_url``` again.

Thank you.

File: lib/carrierwave/mounter.rb (hunk 1 above)
Comment (YotaroSeki) [reply lib/carrierwave/mounter.rb]: I overlooked it.
Thank you.

File: lib/carrierwave/mounter.rb [hunk 3]
```diff
@@ -60,7 +60,7 @@ def cache_names
     end
 
     def cache_names=(cache_names)
-      return if not cache_names or cache_names == "" or uploaders.any?(&:cached?)
+      return if !cache_names || cache_names.blank? || uploaders.any?(&:cached?)
```
Comment (mshibuya) [inline lib/carrierwave/mounter.rb]: This also applies to here.
https://github.com/carrierwaveuploader/carrierwave/pull/2388#discussion_r279695554

File: lib/carrierwave/mounter.rb [hunk 4]
```diff
@@ -70,7 +70,7 @@ def cache_names=(cache_names)
     end
 
     def remote_urls=(urls)
-      return if not urls or urls == "" or urls.all?(&:blank?)
+      return if !urls || urls.blank? || urls.all?(&:blank?)
```
Comment (mshibuya) [inline lib/carrierwave/mounter.rb]: And here.

File: lib/carrierwave/mounter.rb (hunk 3 above)
Comment (YotaroSeki) [reply lib/carrierwave/mounter.rb]: Sorry to bother you again.

I fixed them.

**Comment by mshibuya** [comment]
Great, thank you for contribution!

**Pull request merged**